app/src/main/java/net/gsantner/markor/ui/NewFileDialog.java

```diff
@@ -134,26 +154,17 @@ public void onNothingSelected(AdapterView<?> adapterView) {
                     }
 
                     File f = new File(basedir, fileNameEdit.getText().toString() + fileExtEdit.getText().toString());
-                    String templateContents = getTemplateContent(templateSpinner, basedir);
-                    if (shareUtil.isUnderStorageAccessFolder(f)) {
-                        DocumentFile dof = shareUtil.getDocumentFile(f, false);
-                        if (f.exists() && f.length() < 5 && templateContents != null) {
-                            shareUtil.writeFile(f, false, (fileOpened, fos) -> {
-                                try {
-                                    fos.write(templateContents.getBytes());
-                                } catch (Exception ignored) {
-                                }
-                            });
-                        }
-                        callback(dof != null && dof.canWrite(), f);
-                    } else {
-                        boolean touchOk = FileUtils.touch(f);
-                        if (f.exists() && (f.length() < 5 || BuildConfig.IS_TEST_BUILD) && templateContents != null) {
-                            FileUtils.writeFile(f, templateContents);
+                    final byte[] templateContents = getTemplateContent(templateSpinner, basedir, encryptCheckbox.isChecked());
+                    shareUtil.writeFile(f, false, (arg_ok, arg_fos) -> {
```

Comment:
Are you sure that `if (shareUtil.isUnderStorageAccessFolder(f) ` is really obsolete?

Reply:
In my tries it was not required for this case. The main purpose is, detect if writable as is, otherwise fallback to SAF. Which writeFile basically does inside as well.